### PR TITLE
Movie local data source

### PR DESCRIPTION
**Pull request by MiM14**
## Description de la tarea

I have created local data source for movies app

## ¿Cómo se ha implementado?

There are 3 ways to implement local through a repository:

### XML
The Xml part, It works with Shared Preference and KSerializer, a generic* that onlyuse Gson, but if for some reason I need to implement ,for example Moshi, I could do it changing MovieFactory instead of whatever named gson in the code

### DB
On Db, created with room. It require 5 files to work, define the database, methods, mapper, entity and dao.

### Mem
Memory works with a generic* using datastore, 

## Keywords

Gson, Room, Shared Preferences, DataStore

*Generics, make the code easily abstracted and avoid redundancy between files
In a future pull request I will implement one of these data sources.

### Test
image
XML
image
DB
image
MEM
image

### CODE REVIEW

File: app/src/main/java/com/moaimar/movie_2022/data/local/db/MovieLocalDbDataSource.kt [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+        movies.forEach{
+            dao.save(it.toEntity())
+        }
+    }
+
+    override fun getMoviesFeed():List<Movie>{
+        val movies = dao.getMoviesFeed().map{
+            it.toDomain()
+        }
+    return movies
```
Comment (jmperezra) [inline app/src/main/java/com/moaimar/movie_2022/data/local/db/MovieLocalDbDataSource.kt]: ctrl + alt + L para formatear el código, hazlo siempre.

File: app/src/main/java/com/moaimar/movie_2022/data/local/xml/MovieLocalXmlDataSource.kt [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    override fun getMoviesFeed(): List<Movie> {
+        val movies: MutableList<Movie> = mutableListOf()
+        sharedPref.all.map{
+            movies.add(kSerializer.fromJson(it.value as String, Movie::class.java))
+        }
+        return movies
+    }
+
+    override fun getMovieDetail(movieId: String)=
+        sharedPref.getString(movieId, "null")?.let {
```
Comment (jmperezra) [inline app/src/main/java/com/moaimar/movie_2022/data/local/xml/MovieLocalXmlDataSource.kt]: "null" con comillas ?? 🤔

**Review by jmperezra — CHANGES_REQUESTED** [review]
Revisa los comentarios please, a ver si crees oportuno cambiar algo. Ademas, ten en cuenta que esta PR hace muchas cosas a la vez. Que pasa si pienso que Mem y Xml estan ok pero no Db?

File: app/src/main/java/com/moaimar/movie_2022/data/local/xml/MovieLocalXmlDataSource.kt (hunk 2 above)
Comment (MiM14) [reply app/src/main/java/com/moaimar/movie_2022/data/local/xml/MovieLocalXmlDataSource.kt]: Estuve probando y se me olvido retirar las comillas. Gracias por las observaciones, ya están corregidas

Prueba en cuestión:

`sharedPref.getString(movieId, "null").let {
            kSerializer.fromJson(it, Movie::class.java)!!}`

**Review by jmperezra — APPROVED** [review]
Genial Ismael! 🚀

**Pull request merged**